src/domains/shielder/stores/getShielderIndexedDB.ts

```diff
@@ -1,164 +1,202 @@
+  fee: z.object({
+    amount: z.bigint(),
+    address: z.union([ethAddress, z.literal('native')]),
+  }).optional(),
+});
+
+const partialLocalShielderActivityHistorySchema = localShielderActivityHistorySchema.partial();
+
+const toActivityHistoryStorageFormat = (activities: LocalShielderActivityHistoryArray) =>
+  activities.map(a => ({
+    ...a,
+    amount: a.amount?.toString(),
+    block: a.block?.toString(),
+    relayerFee: a.relayerFee?.toString(),
+    pocketMoney: a.pocketMoney?.toString(),
+    fee: a.fee ? { ...a.fee, amount: a.fee.amount.toString() } : undefined,
   }));
 
-const fromStorageFormat = (data: unknown): Transactions =>
-  z.array(
-    transactionSchema.extend({
-      amount: z.string().transform(BigInt),
-      block: z.string().transform(BigInt),
-      relayerFee: z.string().transform(BigInt).optional(),
-      pocketMoney: z.string().transform(BigInt).optional(),
-    })
-  ).parse(data);
+const fromActivityHistoryStorageFormat = (data: unknown): LocalShielderActivityHistoryArray => {
+  const parsed = z
+    .array(
+      partialLocalShielderActivityHistorySchema.extend({
+        localId: z.string().optional(),
+        amount: z.string().transform(BigInt).optional(),
+        block: z.string().transform(BigInt).optional(),
+        relayerFee: z.string().transform(BigInt).optional(),
+        pocketMoney: z.string().transform(BigInt).optional(),
+        fee: z.object({
+          amount: z.string().transform(BigInt),
+          address: z.union([
+            ethAddress,
+            z.literal('native'),
+          ]),
+        }).optional(),
+        to: ethAddress.optional(),
+        txHash: txHash.optional(),
+      }),
+    )
+    .parse(data);
+
+  return parsed.filter(item => item.localId ?? item.txHash) as LocalShielderActivityHistoryArray;
+};
 
 type DBSchema = {
-  [STORE_CLIENTS]: {
-    key: string,
-    value: ShielderClientData,
-  },
-  [STORE_TRANSACTIONS]: {
-    key: string,
-    value: TransactionsByChain,
-  },
+  [STORE_CLIENTS]: { key: string, value: ShielderClientData },
+  [STORE_LOCAL_SHIELDER_ACTIVITY_HISTORY]: { key: string, value: LocalShielderActivityHistoryByChain },
 };
 
-const initDB = async (): Promise<IDBPDatabase<DBSchema>> => {
+const createDB = (): Promise<IDBPDatabase<DBSchema>> => {
   return openDB(DB_NAME, DB_VERSION, {
     upgrade: db => {
-      if (!db.objectStoreNames.contains(STORE_CLIENTS)) {
-        db.createObjectStore(STORE_CLIENTS);
+      if (db.objectStoreNames.contains(STORE_CLIENTS)) {
+        db.deleteObjectStore(STORE_CLIENTS);
       }
-      if (!db.objectStoreNames.contains(STORE_TRANSACTIONS)) {
-        db.createObjectStore(STORE_TRANSACTIONS);
+      if (db.objectStoreNames.contains(STORE_LOCAL_SHIELDER_ACTIVITY_HISTORY)) {
+        db.deleteObjectStore(STORE_LOCAL_SHIELDER_ACTIVITY_HISTORY);
       }
+
+      db.createObjectStore(STORE_CLIENTS);
+      db.createObjectStore(STORE_LOCAL_SHIELDER_ACTIVITY_HISTORY);
     },
   });
 };
```

Comment:
can we do a similar patters as here, were one module level db Promise is created and awaited wherever needed?
https://github.com/jakearchibald/idb?tab=readme-ov-file#keyval-store

```ts
const dbp = createDB();

...

    getItem: async (itemKey: string): Promise<string | null> => {
      const db = await dbp;
```